src/components/layout/Header.js

```diff
@@ -17,26 +24,62 @@ import {
   STUDENT_MODES,
   TEACHER_MODES,
 } from '../../config/settings';
-import { DEFAULT_VIEW, DASHBOARD_VIEW } from '../../config/views';
+import {
+  DEFAULT_VIEW,
+  DASHBOARD_VIEW,
+  FEEDBACK_VIEW,
+} from '../../config/views';
 import './Header.css';
-import { DEFAULT_PROGRAMMING_LANGUAGE } from '../../config/programmingLanguages';
+import {
+  runCode,
+  openInputSettings,
+  closeInputSettings,
+  patchAppInstanceResource,
+  postAppInstanceResource,
+} from '../../actions';
+import { FEEDBACK, INPUT, STDIN } from '../../config/appInstanceResourceTypes';
+import {
+  JAVASCRIPT,
+  DEFAULT_PROGRAMMING_LANGUAGE,
+} from '../../config/programmingLanguages';
 
 class Header extends Component {
   static propTypes = {
     t: PropTypes.func.isRequired,
+    dispatchPostAppInstanceResource: PropTypes.func.isRequired,
+    dispatchPatchAppInstanceResource: PropTypes.func.isRequired,
+    dispatchRunCode: PropTypes.func.isRequired,
+    dispatchOpenInputSettings: PropTypes.func.isRequired,
+    dispatchCloseInputSettings: PropTypes.func.isRequired,
     classes: PropTypes.shape({
       root: PropTypes.string,
       grow: PropTypes.string,
       logo: PropTypes.string,
     }).isRequired,
-    programmingLanguage: PropTypes.string.isRequired,
     mode: PropTypes.string,
+    currentCode: PropTypes.string.isRequired,
+    currentInput: PropTypes.string,
+    programmingLanguage: PropTypes.string.isRequired,
+    savedCode: PropTypes.string,
+    savedInput: PropTypes.string,
+    inputResourceId: PropTypes.string,
+    stdinResourceId: PropTypes.string,
+    userId: PropTypes.string,
     view: PropTypes.string,
+    feedback: PropTypes.string,
+    isInputDisplayed: PropTypes.bool.isRequired,
```

Comment:
Usually we do not prepend `is` to booleans, so it should be `inputDisplayed`.

Reply:
DONE.